woocommerce/woocommerce-gateway-stripe
# Display the WeChat Pay icon on the legacy checkout

Fixes #3466

## Changes proposed in this Pull Request:

This is small PR fixes an issue that caused the WeChat Pay icon to not be displayed on the legacy/shortcode checkout.

| Before | After |
|--------|--------|
| <img width="771" alt="Screenshot 2024-09-26 at 2 20 01 pm" src="https://github.com/user-attachments/assets/5a022e87-080c-48dd-a4de-b150a66c4236"> | <img width="761" alt="Screenshot 2024-09-26 at 2 16 31 pm" src="https://github.com/user-attachments/assets/89b17531-0c5f-4acc-9d8b-4f4a4b1eba8d"> | 

I noticed while looking at the changes in #3449 that there was an outlier to this list of Payment method icons that WeChat Pay didn't follow the same format. Based on that, I guessed that there must have been an issue with fetching the WeChat Pay icon unless there was code that mapped the `wechat_pay` gateway to use `wechat`, [like there is for cards](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/8.7.0/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L272). I found that there wasn't and therefore there would be an issue fetching the icon dynamically [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/8.7.0/includes/payment-methods/class-wc-stripe-upe-payment-method.php#L216). 

## Testing instructions

1. From your Stripe Payment method settings.
2. Enable WeChat Pay
3. Go to a legacy/shortcode checkout
   - On `develop` there won't be a WeChat icon displayed.
   - On this branch there should be an icon shown. 

---


**Post merge**
